urllib3/response.py

```diff
@@ -327,9 +360,15 @@ def read(self, amt=None, decode_content=None, cache_content=False):
                     # no harm in redundantly calling close.
                     self._fp.close()
                     flush_decoder = True
+                    if (self.strict_content_length and self.length not in (0, None)):
+                        # httplib requires an iterator/string instead of count of bytes
+                        data = [0 for x in range(self._fp_bytes_read)]
```

Comment:
We might also create an empty shell object with the `__len__` value hardcoded to `self._fp_bytes_read` to avoid creating a possibly gigantic list on large streams? That's also an ugly solution.

Maybe the simplest thing is to just create a local, similarly or identically named, exception that takes two ints instead?